core/admin/mailu/sso/views/base.py

```diff
@@ -50,11 +74,22 @@ def login():
             utils.limiter.rate_limit_user(username, client_ip, device_cookie, device_cookie_username) if models.User.get(username) else utils.limiter.rate_limit_ip(client_ip)
             flask.current_app.logger.warn(f'Login failed for {username} from {client_ip}.')
             flask.flash('Wrong e-mail or password', 'error')
-    return flask.render_template('login.html', form=form, fields=fields)
+    return flask.render_template('login.html', form=form, fields=fields, openId=app.config['OIDC_ENABLED'], openIdEndpoint=utils.oic_client.get_redirect_url())
 
 @sso.route('/logout', methods=['GET'])
 @access.authenticated
 def logout():
+    if utils.oic_client.is_enabled():
+        if 'openid_token' not in flask.session:
+            return logout_legacy()
+        if 'state' in flask.request.args and 'state' in flask.session:
+            if flask.args.get('state') == flask.session['state']:
+                logout_legacy()
+        return redirect(utils.oic_client.logout())
+    return logout_legacy()
+    
+
+def logout_legacy():
```

Comment:
Again, I have a problem with the terminology chosen here. "legacy" isn't appropriate.

Reply:
done